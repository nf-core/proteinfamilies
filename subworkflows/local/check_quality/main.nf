/*
    AMINO ACID SEQUENCE QUALITY CHECK
*/

include { SEQKIT_STATS        } from '../../../modules/nf-core/seqkit/stats/main'
include { SEQKIT_STATS_TO_MQC } from '../../../modules/local/seqkit_stats_to_mqc/main'

workflow CHECK_QUALITY {
    take:
    fasta // tuple val(meta), path(fasta)
    
    main:
    ch_versions = Channel.empty()

    SEQKIT_STATS( fasta )
    ch_versions = ch_versions.mix( SEQKIT_STATS.out.versions )

    SEQKIT_STATS_TO_MQC( SEQKIT_STATS.out.stats )
    ch_versions = ch_versions.mix( SEQKIT_STATS_TO_MQC.out.versions )

    emit:
    versions         = ch_versions
    seqkit_stats_mqc = SEQKIT_STATS_TO_MQC.out.mqc
}
