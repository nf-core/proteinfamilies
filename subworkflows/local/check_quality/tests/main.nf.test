nextflow_workflow {

    name "Test Subworkflow CHECK_QUALITY"
    script "../main.nf"
    workflow "CHECK_QUALITY"

    test("fasta") {

        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id: 'test' ],
                    file(params.pipelines_testdata_base_path + 'test_data/mgnifams_input_small.fa', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    workflow.out.seqkit_stats_mqc,
                    workflow.out.versions.collect { path(it).yaml }.unique()
                    ).match()}
            )
        }
    }

}
