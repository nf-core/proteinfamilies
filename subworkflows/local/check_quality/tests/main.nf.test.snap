{
    "faa - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub_after_mqc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub_before_mqc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,0b51b607226ce4fdddba0f37cd0e8e4f",
                    "versions.yml:md5,41d53cf3c37b136275f29fde4e30310c",
                    "versions.yml:md5,5a99fb56bd8aa89101e7e813a2d5a2ea",
                    "versions.yml:md5,8220b546bc6e8498dfb50157f483168b",
                    "versions.yml:md5,be78f4b71c966aed72aa8915c14b94bd"
                ],
                "fasta": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "multiqc_files": [
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub_after_mqc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ],
                    [
                        {
                            "id": "test_stub"
                        },
                        "test_stub_before_mqc.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,0b51b607226ce4fdddba0f37cd0e8e4f",
                    "versions.yml:md5,41d53cf3c37b136275f29fde4e30310c",
                    "versions.yml:md5,5a99fb56bd8aa89101e7e813a2d5a2ea",
                    "versions.yml:md5,8220b546bc6e8498dfb50157f483168b",
                    "versions.yml:md5,be78f4b71c966aed72aa8915c14b94bd"
                ]
            },
            [
                {
                    "CHECK_QUALITY:SEQFU_STATS_BEFORE": {
                        "seqfu": "1.20.3"
                    }
                },
                {
                    "CHECK_QUALITY:SEQFU_STATS_AFTER": {
                        "seqfu": "1.20.3"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_SEQ": {
                        "seqkit": "v2.9.0"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_REPLACE": {
                        "seqkit": "2.9.0"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_RMDUP": {
                        "seqkit": "v2.9.0"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-24T13:07:09.806064938"
    },
    "faa": {
        "content": [
            [
                [
                    {
                        "id": "test_faa"
                    },
                    "test_faa.fasta:md5,242b2a540b013689b51f8814fab92817"
                ]
            ],
            [
                [
                    {
                        "id": "test_faa"
                    },
                    "test_faa_after_mqc.txt:md5,df3ef98dfd4bb4e3af1a993b005c27d8"
                ],
                [
                    {
                        "id": "test_faa"
                    },
                    "test_faa_before_mqc.txt:md5,1c0c35b4d944b5bc09f4e23a1e0faebc"
                ]
            ],
            [
                {
                    "CHECK_QUALITY:SEQFU_STATS_BEFORE": {
                        "seqfu": "1.20.3"
                    }
                },
                {
                    "CHECK_QUALITY:SEQFU_STATS_AFTER": {
                        "seqfu": "1.20.3"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_SEQ": {
                        "seqkit": "v2.9.0"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_REPLACE": {
                        "seqkit": "2.9.0"
                    }
                },
                {
                    "CHECK_QUALITY:SEQKIT_RMDUP": {
                        "seqkit": "v2.9.0"
                    }
                }
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-09-24T13:07:00.378564521"
    }
}